### PR TITLE
Feature iteration 3

### PR DESCRIPTION
**Pull request by lbmerchant93**
### What does this PR do?
Create Sleep, SleepRepo, Activity, and ActivityRepo. Create tests for each, pass those tests.

### Is this a feature or a fix? What does it feature/fix?
Feature, creates functionality for Sleep, SleepRepo, Activity, and ActivityRepo.

### Where should the reviewer start?
Line 1 for:
Sleep
SleepRepo
Activity
ActivityRepo
Sleep-test
SleepRepo-test
Activity-test
ActivityRepo-test

### How should this be tested?
Run the files against its test

### Any background context you want to provide?
no

### Questions:
no

### CODE REVIEW

File: src/ActivityRepo.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  calcAverageMinActiveForAWeek(user, date) {
+    let userActivities = this.allActivities.filter(activity => activity.userID === user.id);
+    let desiredDateIndex = userActivities.map(activity => activity.date).indexOf(date);
+    let desiredWeek = userActivities.slice(desiredDateIndex - 6, desiredDateIndex + 1);
+    let totalTimeActive = desiredWeek.reduce((totalMin, activity) => {
+      totalMin += activity.minutesActive;
+      return totalMin;
+    }, 0);
+    return parseFloat((totalTimeActive / 7).toFixed(1));
+  }
```
Comment (collinkallery) [inline src/ActivityRepo.js:23]: Really awesome job using all of these iterator methods - those are hard to pick up but they are well-executed!

File: src/SleepRepo.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    let desiredWeek = userSleeps.slice(desiredDateIndex - 6, desiredDateIndex + 1);
+    let weekTotal =  desiredWeek.reduce((weekTotal, sleep) => {
+      weekTotal += sleep.sleepQuality;
+      return weekTotal;
+    }, 0);
+    return parseFloat((weekTotal / 7).toFixed(1));
+
+  }
+
+  findAllUsersOverThree(userRepo, date) {
```
Comment (collinkallery) [inline src/SleepRepo.js:78]: The names of these two methods are a little bit vague for a outside developer to read. Maybe have a more descriptive name here (I immediately thought it was checking for all users over the age of three, lol).

**Review by collinkallery — APPROVED** [review]
This looks great, really just some minor suggestions but nothing huge. One thing my partner told me one time that's really helped me with testing is that an outside developer getting to know a code base should theoretically be able to head to the testing suite, read all of the tests, and entirely understand what the codebase does (which I think y'all are doing well here). It is exhausting and inefficient to try to understand a huge code base by literally reading the code itself, so the testing suite should be a high-level guide of what the code is supposed to do. Awesome job!

**Review by mattdeann — APPROVED** [review]
Looks good!

**Pull request merged**